### DISCUSSION TITLE
Custom field transform

### DISCUSSION THREAD

**Discussion by grefrit** [directus/directus#8723] [Archive]
Hi. Is it possible to somehow add a custom field transform? We store the uuid in binary 16 and the api is not working because of this. Is it possible to customize this somehow? Convert uuid to binary 16 and back for any database operation? I tried using hook items.read for this, but that only fixes the display in the collection instead of [objectObject]. I cannot change the query.filter with this (which is correct since items.read does not support .before)

**Comment by rijkvanzanten** [comment]
There's currently no way to inject custom field transform flags. 

@nickrum Could be a good target for another init hook. Moving this bit:

https://github.com/directus/directus/blob/aff49835eeae46a50cf4429f6ba9382b1c81ed59/api/src/services/payload.ts#L49

to an async "getTransformers" function combined with an init hook would allow for registering custom ones
